### PR TITLE
Relicense Halo 2 crates as MIT OR Apache 2.0

### PR DESCRIPTION
**Pull request by str4d**
See [this blog post](https://electriccoin.co/blog/zero-knowledge-proving-system-halo-now-licensed-under-mit-making-it-available-for-anyone-to-use/) for details.

PR cannot be merged without an approving review from an ECC officer with the authority to relicense. For avoidance of ambiguity, the relicense is not complete until this PR is merged.

### CODE REVIEW

File: LICENSE-MIT [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2020-2022 The Electric Coin Company
```
Comment (daira) [inline LICENSE-MIT:3]: Add `Copyright (c) 2022 The halo2 developers`.

**Review by ebfull — APPROVED** [review]

**Review by daira — APPROVED** [review]
utACK 0c50500a4f252d282a0f9502d39549b6b8330eb8
(Note that I don't have authority for the relicensing.)

**Review by steven-ecc — APPROVED** [review]
ACK

**Comment by daira** [comment]
💖🐱🎉

**Pull request merged**